### PR TITLE
[zh-cn] Add downward-api.md chinese version

### PR DESCRIPTION
**Pull request by zhangxyjlu**
fixed: #34576 
Add downward-api.md chinese version

### CODE REVIEW

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+**downward API** 允许容器在不使用 Kubernetes 客户端或 API 服务器的情况下使用自己或集群的信息。
+
+<!--
+An example is an existing application that assumes a particular well-known
+environment variable holds a unique identifier. One possibility is to wrap the
+application, but that is tedious and error-prone, and it violates the goal of low
+coupling. A better option would be to use the Pod's name as an identifier, and
+inject the Pod's name into the well-known environment variable.
+-->
+例如，现有应用程序假设特定的已知环境变量具有唯一标识符。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:35]:
```suggestion
例如，现有应用程序假设某特定的周知的环境变量是存在的，其中包含唯一标识符。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+一种方法是包装应用程序，但这很繁琐且容易出错，并且违背了低耦合的目标。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:36]:
```suggestion
一种方法是对应用程序进行封装，但这很繁琐且容易出错，并且违背了低耦合的目标。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+更好的选择是使用 Pod 名称作为标识符，并将 Pod 名称注入到公共的环境变量中。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:37]:
```suggestion
更好的选择是使用 Pod 名称作为标识符，并将 Pod 名称注入到周知的环境变量中。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+<!--
+In Kubernetes, there are two ways to expose Pod and container fields to a running container:
+
+* as [environment variables](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api)
+* as [files in a `downwardAPI` volume](/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)
+-->
+在 Kubernetes 中，有两种方法可以将 Pod 和容器字段暴露给正在运行的容器：
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:45]:
```suggestion
在 Kubernetes 中，有两种方法可以将 Pod 和容器字段暴露给运行中的容器：
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+* 作为[环境变量](/zh-cn/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api)
+* 作为 [`downwardAPI` 卷中的文件](/zh-cn/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)
+
+<!--
+Together, these two ways of exposing Pod and container fields are called the
+_downward API_.
+-->
+这两种暴露 Pod 和容器字段的方式统称为 **downward API**。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:54]:
```suggestion
这两种暴露 Pod 和容器字段的方式统称为 **Downward API**。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+<!--
+## Available fields
+
+Only some Kubernetes API fields are available through the downward API. This
+section lists which fields you can make available.
+-->
+## 可用字段  {#available-fields}
+
+只有部分 Kubernetes API 字段可以通过 downward API 使用。本节列出了你可以使用的字段。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:66]:
```suggestion
只有部分 Kubernetes API 字段可以通过 Downward API 使用。本节列出了你可以使用的字段。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+<!--
+You can pass information from available Pod-level fields using `fieldRef`.
+At the API level, the `spec` for a Pod always defines at least one
+[Container](/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container).
+You can pass information from available Container-level fields using
+`resourceFieldRef`.
+-->
+你可以使用 `fieldRef` 从可用的 Pod 级字段中传递信息。在 API 层面，一个 Pod 的
+`spec` 总是定义了至少一个 [Container](/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:76]: 尽管还没翻译，也可以将链接先改成中文。

```suggestion
`spec` 总是定义了至少一个 [Container](/zh-cn/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+### 通过 `fieldRef` 获得信息  {#downwardapi-fieldRef}
+
+对于大多数 Pod 级别的字段，你可以将它们作为环境变量或使用 `downwardAPI` 卷提供给容器。
+通过这两种机制可用的字段有：
+
+`metadata.name`
+<!--
+: the pod's name
+-->
+Pod 的名称
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:95]: 这里的处理都是错的。原来的英文表达是 definition 的结构。https://www.markdownguide.org/extended-syntax/#definition-lists

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+<!--
+: A container's CPU request
+-->
+容器的 CPU 请求值
+
+`resource: limits.memory`
+<!--
+: A container's memory limit
+-->
+容器的内存约束值
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:188]:
```suggestion
容器的内存限制值
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+[requests and limits](/docs/concepts/configuration/manage-resources-containers/#requests-and-limits)
+for resources such as CPU and memory.
+-->
+### 通过 `resourceFieldRef` 获得信息  {#downwardapi-resourceFieldRef}
+
+`resource: limits.cpu`
+<!--
+: A container's CPU limit
+-->
+容器的 CPU 约束值
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:176]:
```suggestion
容器的 CPU 限制值
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+: A container's hugepages request (provided that the `DownwardAPIHugePages` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled)
+-->
+容器的巨页请求值（前提是启用了 `DownwardAPIHugePages`
+[特性门控](/zh-cn/docs/reference/command-line-tools-reference/feature-gates/)）
+
+`resource: limits.ephemeral-storage`
+<!--
+: A container's ephemeral-storage limit
+-->
+容器的临时存储约束值
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:214]:
```suggestion
容器的临时存储的限制值
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+If CPU and memory limits are not specified for a container, and you use the
+downward API to try to expose that information, then the
+kubelet defaults to exposing the maximum allocatable value for CPU and memory
+based on the [node allocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
+calculation.
+-->
+#### 资源限制的后备信息  {#fallback-information-for-resource-limits}
+
+如果没有为容器指定 CPU 和内存限制，并且使用 downward API 试图暴露该信息，那么 kubelet 默认会根据
+[node allocatable](/zh-cn/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:234]:
```suggestion
[节点可分配资源](/zh-cn/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+计算暴露 CPU 和内存的最大可分配值。
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:235]:
```suggestion
计算并暴露 CPU 和内存的最大可分配值。
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+
+If CPU and memory limits are not specified for a container, and you use the
+downward API to try to expose that information, then the
+kubelet defaults to exposing the maximum allocatable value for CPU and memory
+based on the [node allocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
+calculation.
+-->
+#### 资源限制的后备信息  {#fallback-information-for-resource-limits}
+
+如果没有为容器指定 CPU 和内存限制，并且使用 downward API 试图暴露该信息，那么 kubelet 默认会根据
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:233]:
```suggestion
如果没有为容器指定 CPU 和内存限制，并且使用 Downward API 试图暴露该信息，那么 kubelet 默认会根据
```

File: content/zh-cn/docs/reference/glossary/downward-api.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+-->
+有时，容器拥有关于自身的信息是很有用的，而不需要对容器代码进行更改，从而使其直接耦合到 Kubernetes。
+
+<!--
+The Kubernetes downward API allows containers to consume information about themselves
+or their context in a Kubernetes cluster. Applications in containers can have
+access to that information, without the application needing to act as a client of
+the Kubernetes API.
+-->
+Kubernetes downward API 允许容器在 Kubernetes 集群中使用它们自己或它们上下文的信息。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/glossary/downward-api.md:40]:
```suggestion
Kubernetes Downward API 允许容器在 Kubernetes 集群中使用它们自己或它们上下文的信息。
```

File: content/zh-cn/docs/reference/glossary/downward-api.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+- using [a `downwardAPI` volume](/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)
+
+Together, these two ways of exposing Pod and container fields are called the _downward API_.
+-->
+有两种方法可以将 Pod 和容器字段暴露给正在运行的容器：
+
+* 使用[环境变量](/zh-cn/docs/tasks/inject-data-application/environment-variable-expose-pod-information/)
+* 使用 [`downwardAPI` 卷](/zh-cn/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)
+
+这两种暴露 Pod 和容器字段的方式统称为 **downward API**。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/glossary/downward-api.md:56]:
```suggestion
这两种暴露 Pod 和容器字段的方式统称为 **Downward API**。
```

**Comment by zhangxyjlu** [comment]
I've revised all your reviews，thanks@[tengqm](https://github.com/tengqm)

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 17]
```diff
@@ -0,0 +1,270 @@
...
+a `downwardAPI` volume `fieldRef`, but **not as environment variables**:
+-->
+此外，以下信息可以通过 `downwardAPI` 卷 `fieldRef` 获得，但**不能作为环境变量**获得：
+
+<!--
+`metadata.labels`
+: all of the pod's labels, formatted as `label-key="escaped-label-value"` with one label per line
+-->
+`metadata.labels`
+: 所有 Pod 的标签，格式为 `label-key="escaped-label-value"`，每行一个标签
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:165]:
```suggestion
: Pod 的所有标签，格式为 `标签键名="转义后的标签值"`，每行一个标签
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 18]
```diff
@@ -0,0 +1,270 @@
...
+
+<!--
+`metadata.annotations`
+: all of the pod's annotations, formatted as `annotation-key="escaped-annotation-value"` with one annotation per line  
+-->
+`metadata.annotations`
+: 所有 Pod 的注解，格式为 `annotation-key="escaped-annotation-value"`，每行一个注解
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:172]:
```suggestion
: Pod 的全部注解，格式为 `注解键名="转义后的注解值"`，每行一个注解
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 19]
```diff
@@ -0,0 +1,270 @@
...
+
+<!--
+### Information available via `resourceFieldRef` {#downwardapi-resourceFieldRef}
+
+These container-level fields allow you to provide information about
+[requests and limits](/docs/concepts/configuration/manage-resources-containers/#requests-and-limits)
+for resources such as CPU and memory.
+-->
+### 通过 `resourceFieldRef` 获得信息  {#downwardapi-resourceFieldRef}
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:181]:
```suggestion
### 可通过 `resourceFieldRef` 获得的信息  {#downwardapi-resourceFieldRef}
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 20]
```diff
@@ -0,0 +1,270 @@
...
+
+If CPU and memory limits are not specified for a container, and you use the
+downward API to try to expose that information, then the
+kubelet defaults to exposing the maximum allocatable value for CPU and memory
+based on the [node allocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable)
+calculation.
+-->
+#### 资源限制的后备信息  {#fallback-information-for-resource-limits}
+
+如果没有为容器指定 CPU 和内存限制，并且使用 Downward API 试图暴露该信息，那么 kubelet 默认会根据
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:252]:
```suggestion
如果没有为容器指定 CPU 和内存限制时尝试使用 Downward API 暴露该信息，那么 kubelet 默认会根据
```

File: content/zh-cn/docs/concepts/workloads/pods/downward-api.md [hunk 21]
```diff
@@ -0,0 +1,270 @@
...
+<!--
+You can read about [`downwardAPI` volumes](/docs/concepts/storage/volumes/#downwardapi).
+
+You can try using the downward API to expose container- or Pod-level information:
+* as [environment variables](/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api)
+* as [files in `downwardAPI` volume](/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)
+-->
+你可以阅读有关 [`downwardAPI` 卷](/zh-cn/docs/concepts/storage/volumes/#downwardapi)的内容。
+
+您可以尝试使用 Downward API 暴露容器或 Pod 级别的信息：
```
Comment (tengqm) [inline content/zh-cn/docs/concepts/workloads/pods/downward-api.md:268]:
```suggestion
你可以尝试使用 Downward API 暴露容器或 Pod 级别的信息：
```

File: content/zh-cn/docs/reference/glossary/downward-api.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+<!--
+Kubernetes' mechanism to expose Pod and container field values to code running in a container.
+-->
+Kubernetes 将 Pod 和容器字段值暴露给容器中运行的代码的机制。
+<!--more-->
+<!--
+It is sometimes useful for a container to have information about itself, without
+needing to make changes to the container code that directly couple it to Kubernetes.
+-->
+有时，容器拥有关于自身的信息是很有用的，而不需要对容器代码进行更改，从而使其直接耦合到 Kubernetes。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/glossary/downward-api.md:32]:
```suggestion
有时，在不需要修改容器代码的前提下让容器拥有关于自身的信息是很有用的。修改代码可能使容器直接耦合到 Kubernetes。
```

File: content/zh-cn/docs/reference/glossary/downward-api.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+<!--
+The Kubernetes downward API allows containers to consume information about themselves
+or their context in a Kubernetes cluster. Applications in containers can have
+access to that information, without the application needing to act as a client of
+the Kubernetes API.
+-->
+Kubernetes Downward API 允许容器在 Kubernetes 集群中使用它们自己或它们上下文的信息。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/glossary/downward-api.md:40]:
```suggestion
Kubernetes Downward API 允许容器使用它们自己或它们在 Kubernetes 集群中所处环境的信息。
```

File: content/zh-cn/docs/reference/glossary/downward-api.md [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+容器中的应用程序可以访问该信息，而不需要通过 Kubernetes API 客户端。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/glossary/downward-api.md:41]:
```suggestion
容器中的应用程序可以访问该信息，而不需要以 Kubernetes API 客户端的形式执行操作。
```

**Comment by zhangxyjlu** [comment]
Thank you for your guidance@[tengqm](https://github.com/tengqm)

**Review by ydFu — APPROVED** [review]
Looks good to me,thank.

/lgtm

**Pull request merged**